examples/objc/main_darwin.go

```diff
@@ -17,45 +17,42 @@ var (
 	sel_bar    = objc.RegisterName("bar")
 )
 
-type barObject struct {
-	isa objc.Class `objc:"BarObject : NSObject <NSDelegateWindow>"`
-	bar int
-}
-
-func (b *barObject) Init(_cmd objc.SEL) objc.ID {
-	return objc.ID(unsafe.Pointer(b)).SendSuper(_cmd)
-}
-
-func (b *barObject) Bar(_cmd objc.SEL) int {
-	return b.bar
-}
-
-func (b *barObject) SetBar(_cmd objc.SEL, bar int) {
-	b.bar = bar
-}
-
-func (_ *barObject) Selector(metName string) objc.SEL {
-	switch metName {
-	case "Init":
-		return sel_init
-	case "SetBar":
-		return sel_setBar
-	case "Bar":
-		return sel_bar
-	default:
-		return 0
-	}
+func BarInit(id objc.ID, _cmd objc.SEL) objc.ID {
+	return id.SendSuper(_cmd)
 }
 
 func main() {
-	class, err := objc.RegisterClass(&barObject{})
+	/*
+		This struct is equivalent to the following Objective-C definition.
+
+		@interface BarObject : NSObject <NSDelegateWindow>
+		@property (readwrite) bar int
+		@end
+	*/
+	class, err := objc.RegisterClass(
+		"BarObject",
+		objc.GetClass("NSObject"),
+		[]*objc.Protocol{
+			objc.GetProtocol("NSDelegateWindow"),
+		},
+		[]objc.FieldDef{
+			{
+				Name:      "bar",
+				Type:      reflect.TypeOf(int(0)),
```

Comment:
Should the Type field just be an interface instead of reflect.Type and then we just obtain the type using reflection? That way there is no dependency on the caller to import reflect.

Reply:
You mean we might want to write like this?

```go
{
    Name: "bar",
    Type: 0,
    Attribute: objc.ReadWrite
}
```

I slightly prefer `reflect.Type` as this is more explicit.